react/modules/tasks/nextjs-ssr-ssg.md

```diff
@@ -11,30 +11,29 @@
    - Create a separate branch for this point from "nextjs-ssr-pages-api" branch. Branch name: "nextjs-ssr-app-router-api".
    - You must use file-based routing provided by next.js (App Router)
    - [Migrate from Pages Api to App Router](https://nextjs.org/docs/pages/building-your-application/upgrading/app-router-migration).
-4. Remix Integration:
-   - Create a separate branch for this point from the previous branch task. Branch name: "remix-ssr".
-   - Add the Remix framework to your vite config
-   - [Update routing](https://remix.run/docs/en/main/file-conventions/routes)
-   - Migrate your application to ssr with Remix
+4. [Optional] React Router 7 Integration:
+   - Create a separate branch for this point from the previous branch task. Branch name: "react-router-ssr".
+   - [Update routing](https://reactrouter.com/start/framework/routing)
+   - Migrate your application to ssr with React Router 7: data fetching should be done via loaders on the server-side
 
 ## Application Requirements
 
 1. Application in each branch should function in accordance to the requirements provided in the previous modules, if they do not contradict with the new requirements provided in this module.
-2. All the tests should pass, test coverage should be >= 80%.
+2. All the tests should pass, test coverage should be >= 70%.
 
 ## Points
 
-A student can achieve a maximum of 150 points.
+A student can achieve a maximum of 100 points (+50 optional).
```

Comment:
What does that "optional" mean? If I didn't implement something or have errors in Next.js Pages API or Next.js App Router API and got for example 80 points and not 100, I can get 100 points with React Router 7, right?

Reply:
Yes, we have a penalty for implementing SSR just using React Router 7 (without implementing NextJS approach). If the NextJS (using pages and app router) had been implemented, student can work on the optional task